news/bld-new-mindeps.rst

```diff
@@ -0,0 +1,6 @@
+Building h5py
+-------------
+
+* The minimum versions build-time Python requirements were updated to
+  ``Cython==3.0.0`` (up from ``0.29.1``), and ``numpy==1.25.0`` (down from
+  ``2.0.0``).
```

Comment:
Let's mention here that we still recommend building with Numpy 2.x in general, this is just to allow ecosystems where you build with the same version you want to use to still use Numpy 1.x.

Reply:
good idea. How's my attempt at it ?